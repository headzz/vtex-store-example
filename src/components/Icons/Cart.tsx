import Image from 'next/image'

/**
 * @component
 * @name Cart
 * @description Defines an image component used when buying things.
 * @namespace components components/Icons/Cart
 *
 */
const Cart: React.FC = () => (
  <Image
    src="/img/icon/cart.svg"
    height={24}
    width={24}
    alt="ícone verificação de itens no carrinho"
  />
)

export default Cart
