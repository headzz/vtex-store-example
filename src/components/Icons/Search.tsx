import Image from 'next/image'

/**
 * @component
 * @name Search
 * @description Defines an image component used in search bar.
 * @namespace components components/Icons/Search
 *
 */
const Search = () => (
  <Image
    src="/img/icon/search.svg"
    height={24}
    width={24}
    alt="Ã­cone busca status"
  />
)

export default Search
