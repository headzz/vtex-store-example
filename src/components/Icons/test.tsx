import { render, screen } from '@testing-library/react'

import Cart from './Cart'
import Search from './Search'

describe('Icons', () => {
  it('should render component Cart icon', () => {
    render(<Cart />)

    expect(
      screen.getByAltText('ícone verificação de itens no carrinho')
    ).toMatchSnapshot()
    expect(
      screen.getByAltText('ícone verificação de itens no carrinho')
    ).toHaveAttribute('alt', 'ícone verificação de itens no carrinho')
  })

  it('should render component Search icon', () => {
    render(<Search />)

    expect(screen.getByAltText('ícone de pesquisa')).toMatchSnapshot()
    expect(screen.getByAltText('ícone de pesquisa')).toHaveAttribute(
      'alt',
      'ícone de pesquisa'
    )
  })
})
