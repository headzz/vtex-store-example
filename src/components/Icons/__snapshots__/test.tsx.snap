// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons should render component Cart icon 1`] = `
<img
  alt="ícone verificação de itens no carrinho"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
/>
`

exports[`Icons should render component Search icon 1`] = `
<img
  alt="ícone de pesquisa"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
/>
`
